spec/fig_offchain_prot.tex

```diff
@@ -93,7 +93,7 @@
 							\Req{$s = \hats + 1 ~ \land ~ \hpLdr(s) = j$} \; %
 
 							% Wait for snapshot no snapshot in flight anymore and all txs applicable
-							\Wait{$\bars = \hats ~ \land ~ \barmU \applytx \mT_{res} \not= \bot$}{ %
+							\Wait{$\bars = \hats ~ \land ~ \barmU \applytx \mT_{res} \not= \bot ~ \and ~ \mT_{res} \subseteq \hatmT$}{ %
```

Comment:
This would not typecheck if we would have written all this in agda :D

I think we need to change the signature of `onReq` to have ${\mathcal T}_{req}^{\#}$ as argument, then have an additional `wait` with $\forall h \in {\mathcal T}_{req}^{\#} : (h, \cdot) \in {\mathcal T}$ and a resolve statement defining ${\mathcal T}_{res}$, i.e. what I had done here (with $\hat {\mathcal T}$ this time though):

https://github.com/input-output-hk/hydra/issues/728#issuecomment-1460156847

Reply:
I'd be happy to pair on this.. and then we see together that we should have written the spec in agda

Reply:
Preaching to the choir :)

Reply:
Yeah so the spec has not been updated in this PR? Do we have a follow-up item which makes us not forget this?